management/src/test/java/org/ehcache/management/providers/statistics/StatsUtil.java

```diff
@@ -120,7 +120,7 @@ public static double getExpectedValueFromRatioHistory(String statName, Context c
         int mostRecentIndex = ratioHistory.getValue().length - 1;
         value = ratioHistory.getValue()[mostRecentIndex].getValue();
       }
-    } while (!Thread.currentThread().isInterrupted() && value != expectedResult);
+    } while (!Thread.currentThread().isInterrupted() && !value.equals(expectedResult));
```

Comment:
If another wait time must be added it matches what we saw in debug with Ludovic: we needed to wait at least for 2 computations of samples to happen before doing any cache ops because the ratio depends on previous stats.

So I think we would only need to wait for the first 2 samples to be computed into `StatsUtil .triggerStatComputation()`

i.e.: replacing

`if (((StatisticHistory<?, ?>) entry.getValue()).getValue().length == 0) {`

by:

`if (((StatisticHistory<?, ?>) entry.getValue()).getValue().length < 2) {`

will perhaps do the trick.

It would wait _before_ starting the test and doing some cache ops, and then, we wouldn't need to sleep in the loop.

`StatsUtil .triggerStatComputation()` is supposed to be the place where we do the necessary stuff to make sure the stats are _ready_.